great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -2634,23 +2633,23 @@ def _spark_column_map_condition_value_counts(
     unexpected_condition, compute_domain_kwargs, accessor_domain_kwargs = metrics.get(
         "unexpected_condition"
     )
-    df = execution_engine.get_domain_records(domain_kwargs=compute_domain_kwargs)
-
     if "column" not in accessor_domain_kwargs:
         raise ValueError(
             """No "column" found in provided metric_domain_kwargs, but it is required for a column map metric
 (_spark_column_map_condition_value_counts).
 """
         )
-
     column_name: Union[str, quoted_name] = accessor_domain_kwargs["column"]
-
     column_name = get_dbms_compatible_column_names(
         column_names=column_name,
         batch_columns_list=metrics["table.columns"],
         execution_engine=execution_engine,
     )
 
+    df: "pyspark.sql.dataframe.DataFrame" = execution_engine.get_domain_records(
+        domain_kwargs=compute_domain_kwargs
+    )
```

Comment:
small refactor: type hinting added and df instantiated closer to where it is first being used.

Reply:
I also like that we now do error checking first.